### PR TITLE
exported KubernetesContainerRunner, KubernetesContainerRunnerOptions,…

### PR DESCRIPTION
**Pull request by matteosilv**
In my previous PR I forgot to export public symbols and generate api-report.md

ref: https://github.com/backstage/backstage/pull/12839

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by matteosilv** [comment]
Can somebody help me understend why e2e tests are failing after I uploaded the api-report.md?

**Review by jhaals — CHANGES_REQUESTED** [review]
Thanks for the PR!
We had some build issues last week and had to force push to the mater branch in order to clear up the issues. Rebase your PR on master and it should hopefully clear up 🤞

**Comment by matteosilv** [comment]
> Thanks for the PR! We had some build issues last week and had to force push to the mater branch in order to clear up the issues. Rebase your PR on master and it should hopefully clear up 🤞

Guys I added @types/js-yaml and @types/request to the backend-common package.json, it fixed a couple of workflows but still some e2e tests are failing because of js-yaml and I don't have any other ideas to fix it. Can you give me a hand?

File: packages/backend-common/api-report.md [hunk 1]
```diff
@@ -456,13 +461,68 @@ export { isChildPath };
...
+export class KubernetesContainerRunner implements ContainerRunner {
+  constructor(options: KubernetesContainerRunnerOptions);
+  // (undocumented)
+  createJob(jobSpec: V1Job): Promise<any>;
+  // (undocumented)
+  getNamespace(kubeConfig: KubeConfig, namespace?: string): string;
+  // (undocumented)
+  handleError(err: any, errorCallback: (reason: any) => void): void;
+  // (undocumented)
+  runContainer(options: RunContainerOptions): Promise<void>;
```
Comment (jhaals) [inline packages/backend-common/api-report.md:474]: This should be the only method that should be made public right?
Would be nice to slim down the API surface of what we export :)

**Review by jhaals — CHANGES_REQUESTED** [review]
Thanks for the PR! Some minor tweaks needed to the API which probably removes the need for those additional dependencies as well :)

File: packages/backend-common/api-report.md [hunk 2]
```diff
@@ -456,13 +458,36 @@ export { isChildPath };
...
+
 // @public
 export function loadBackendConfig(options: {
   logger: Logger;
   remote?: LoadConfigOptionsRemote;
   argv: string[];
 }): Promise<Config>;
 
+// @public
+export type MountBaseOptions = {
```
Comment (jhaals) [inline packages/backend-common/api-report.md:486]:
```suggestion
export type KubernetesContainerRunnerMountBase = {
```

File: .changeset/kind-walls-share.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/backend-common': patch
+---
+
+exported KubernetesContainerRunner, KubernetesContainerRunnerOptions, MountBaseOptions
```
Comment (jhaals) [inline .changeset/kind-walls-share.md:5]:
```suggestion
exported `KubernetesContainerRunner`, `KubernetesContainerRunnerOptions` and `KubernetesContainerRunnerMountBase`
```

**Review by jhaals — CHANGES_REQUESTED** [review]
Thanks for the fixes! Some minor fixes left and then I think we're good to go :)

**Review by jhaals — CHANGES_REQUESTED** [review]
Hi @matteosilv!

It looks like this is due to an underlying issue in the Kubernetes client library for javascript. There's currently an open issue in https://github.com/kubernetes-client/javascript/issues/883 describing the problem. I don't think we want to add these dependencies in the create-app template as it's likely that it might cause issues for users later.
Could we wait with this until we have gotten an update from the kubernetes folks? Most likely this is a mistake that should be reverted.

**Comment by matteosilv** [comment]
> Hi @matteosilv!
> 
> It looks like this is due to an underlying issue in the Kubernetes client library for javascript. There's currently an open issue in [kubernetes-client/javascript#883](https://github.com/kubernetes-client/javascript/issues/883) describing the problem. I don't think we want to add these dependencies in the create-app template as it's likely that it might cause issues for users later. Could we wait with this until we have gotten an update from the kubernetes folks? Most likely this is a mistake that should be reverted.

Absolutely! It seemed a little odd to me too

**Comment by matteosilv** [comment]
Don't know what changed, but taking in recent master branch commits seems to have fixed the issue. I just removed the edits to the create-app `package.json.hbs` template, because I want to be sure they are actually needed, before issuing an un-needed patch version for a module

**Comment by matteosilv** [comment]
Something is moving 😄 https://github.com/kubernetes-client/javascript/pull/890

**Comment by danmontgomery** [comment]
https://github.com/kubernetes-client/javascript/pull/890 was merged so this should be addressed with next release

**Comment by matteosilv** [comment]
> [kubernetes-client/javascript#890](https://github.com/kubernetes-client/javascript/pull/890) was merged so this should be addressed with next release

We need to wait for 0.17.2 and merge after then! Thanks!

**Comment by matteosilv** [comment]
Just wanted to re-open this as we are waiting for javascript kubernetes client  0.17.2 to come out

**Comment by matteosilv** [comment]
There is still no date for 0.17.2...

**Comment by matteosilv** [comment]
This request is stuck since September.... Do we have any alternative to merge it in, without relying on Kubernetes javascript client being updated?

**Comment by matteosilv** [comment]
> Hi @matteosilv! Yeah it's taking a looong time 😩 I asked around and got news that we have pinned the version to 0.17.0 for now which means we shouldn't need those additional dependencies. Could you try rebasing and removing the dependencies. Hopefully it should work! 🤞

Ok trying force pushing the initial commit from scratch

**Comment by matteosilv** [comment]
yeaaaaaahhhh all checks passed. Guess we can finally merge 😄

**Review by jhaals — APPROVED** [review]
@matteosilv thank you for keeping this work alive 👏 🏅

**Comment by matteosilv** [comment]
@danmontgomery the KubernetesContainerRunner is finally released with backstage v1.9.0, package backend-common 0.17.0

**Pull request merged**